lib/models/LifecycleConfiguration.js

```diff
@@ -709,6 +994,40 @@ class LifecycleConfiguration {
                     Action = `<${actionName}>${Days}${Date}${DelMarker}` +
                         `</${actionName}>`;
                 }
+                if (actionName === 'NoncurrentVersionTransition') {
+                    const xml = [];
+                    nonCurrentVersionTransition.forEach(transition => {
+                        const { noncurrentDays, storageClass } = transition;
+                        xml.push(
+                            `<${actionName}>`,
+                                `<NoncurrentDays>${noncurrentDays}` +
+                                '</NoncurrentDays>',
+                                `<StorageClass>${storageClass}</StorageClass>`,
+                            `</${actionName}>`
+                        );
+                    });
+                    Action = xml.join('');
+                }
+                if (actionName === 'Transition') {
+                    const xml = [];
+                    transition.forEach(transition => {
+                        const { days, date, storageClass } = transition;
+                        let element;
+                        if (days !== undefined || days === 0) {
```

Comment:
I think you mean `days !== undefined && days !== 0`, could also be more restrictive with `days > 0`

Reply:
It looks like the comparison to zero is actually redundant here. I'll just remove it altogether because at this point there is no need to check that the number is `>= 0` since the PUT already restricts this.

Reply:
Nice catch!